stix_shifter_modules/onelogin/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,193 @@
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, ComparisonExpression, \
+    ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+from stix_shifter_utils.stix_translation.src.utils.transformers import TimestampToMilliseconds
+from stix_shifter_utils.stix_translation.src.json_to_stix import observable
+import logging
+import re
+
+# Source and destination reference mapping for ip and mac addresses.
+# Change the keys to match the data source fields. The value array indicates the possible data type that can come into from field.
+REFERENCE_DATA_TYPES = {"ipaddr": ["ipv4", "ipv4_cidr", "ipv6", "ipv6_cidr"],
+                        "proxy_ip": ["ipv4", "ipv4_cidr"],
+                        }
+
+logger = logging.getLogger(__name__)
+
+
+class QueryStringPatternTranslator:
+    # Change comparator values to match with supported data source operators
+    comparator_lookup = {
+        ComparisonExpressionOperators.And: "&",
```

Comment:
Is this all the supported comparison operations supported by OneLogin? How are OR conditions handled? You'd also need to at least have something for `ObservationOperators.Or` and `ObservationOperators.And`, both of which would equate to an OR statement (or equivalent) on the native data source query. Without this, any pattern with multiple observations would raise an error.

Reply:
@delliott90 
Okay. 
OneLogin data source only supports '& (AND)' as a comparison operation.
I think we have to equate '&' for both `ObservationOperators.Or` and `ObservationOperators.And` since our data source (OneLogin) does not supporting OR operator while searching events.  
Like :
        ObservationOperators.And: '&',
        ObservationOperators.Or: '&'

Reply:
I think you may need to break up each observation into its own query, so you will return multiple translated queries in the list returned from the query constructor. STIX observations separated by AND must be different observations (ie. events) but with translating the ObservationOperator AND to &, you are looking for one event in OneLogin with both data elements.

Reply:
@delliott90 okay we have made changes as per your suggestion please review.